mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -576,8 +609,7 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     (splits, categoricalFeaturesInfo_, labelType)
   }
 
-  override def runTest(rdd: RDD[LabeledPoint])
-    : Either[RandomForestModel, GradientBoostedTreesModel] = {
+  override def runTest(rdd: RDD[LabeledPoint]): DecisionTreeModel = {
```

Comment:
I think it'd be fairer to create the DataFrame, cache it, and materialize it before calling spark.ml algorithms.